libRALFit/src/ral_nlls_internal.f90

```diff
@@ -4064,12 +4066,20 @@ Recursive subroutine evaltensor_f(status, n, m, s, f, params)
           call calculate_sHs(n,m, s, params)
 
           ! put them all together for the first 1:m terms of f
-          f(1:params%m) = params%f(1:params%m) + params%tenJ%Js(1:params%m) +  &
```

Comment:
@tyronerees this is the proposal for `evaltensor_f`to address slices inside assumed-shape arrays, can you review it? just to make sure everything is right.